content/en/platform/corda/5.2/deploying-operating/deployment/upgrading521/_index.md

```diff
@@ -0,0 +1,517 @@
+{{< /note >}}
+
+```
+docker run -v $(pwd)/kafka_config:/kafka_config -v $(pwd)/sql_updates:/sql_updates \
+--add-host prereqs-kafka:host-gateway --add-host rest-api:host-gateway $CLI_DOCKER_IMAGE \
+upgrade migrate-data-5-2-1 -b=prereqs-kafka:9092 --kafka-config=/kafka_config/props.txt \
+-t https://rest-api:8888 -u admin -p admin --insecure \
+-l /sql_updates
+```
+
+Inspect the generated SQL files. You cannot apply them until database schema changes have been completed in step [Migrate the Corda Cluster Database]({{< relref "#migrate-the-corda-cluster-database" >}}). Keep these SQL files until then.
+
+## Scale Down the Running Corda Worker Instances
+
+{{< important >}}
+After completing this step and until you finish the last step in these instructions, Corda 5.2 becomes unresponsive. To reduce Corda downtime, R3 recommends that you test the database migration and become fully familiar with the next set of steps before proceeding with this one.
+{{< /important >}}
+
+You can scale down the workers using any tool of your choice. For example, run the following commands if using `kubectl`:
+
+```
+kubectl scale --replicas=0 deployment/corda-crypto-worker -n corda
+kubectl scale --replicas=0 deployment/corda-db-worker -n corda
+kubectl scale --replicas=0 deployment/corda-flow-mapper-worker -n corda
+kubectl scale --replicas=0 deployment/corda-flow-worker -n corda
+kubectl scale --replicas=0 deployment/corda-membership-worker -n corda
+kubectl scale --replicas=0 deployment/corda-p2p-gateway-worker -n corda
+kubectl scale --replicas=0 deployment/corda-p2p-link-manager-worker -n corda
+kubectl scale --replicas=0 deployment/corda-persistence-worker -n corda
+kubectl scale --replicas=0 deployment/corda-rest-worker -n corda
+kubectl scale --replicas=0 deployment/corda-token-selection-worker -n corda
+kubectl scale --replicas=0 deployment/corda-uniqueness-worker -n corda
+kubectl scale --replicas=0 deployment/corda-verification-worker -n corda
+```
+
+If you are scripting these commands, you can wait for the workers to be scaled down using something similar to the following:
+
+```
+while [ "$(kubectl get pods --field-selector=status.phase=Running -n corda | grep worker | wc -l | tr -d ' ')" != 0 ]
+do
+sleep 1
+done
+```
+
+## Migrate the Corda Cluster Database
+
+Migrating the database schemas is a four-step process. The first generates SQL scripts, the second applies them to the database. The breaking up of these steps is about transparency, it allows database administrators to review the SQL scripts to ensure they are happy with the content before applying it to their databases. The third step is about granting the necessary permissions to any new tables in the database and is similar to the step run at bootstrapping time as described in [Manual Bootstrapping]({{< relref "../deploying/manual-bootstrapping.md" >}}). Lastly, you must apply the previously-generated SQL files.
```

Comment:
`Migrating the database schemas is a four-step process. The first generates SQL scripts, the second applies them to the database.`

This doesn't really summarise the steps below any more. Perhaps something more vague like `Migrating the database schema requires generating SQL scripts and applying them as separate steps`.